src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java

```diff
@@ -234,6 +234,580 @@ public NonBlockingStatsDClient(final String prefix, final int queueSize, String[
         executor.submit(statsDSender);
     }
 
+    /**
+     * Create a new StatsD client communicating with a StatsD instance on the
+     * specified host and port.
+     * This is a shallow copy constructor meant to be used internally only.
+     *
+     * @param client
+     *    source object to copy
+     */
+    private NonBlockingStatsDClient(NonBlockingStatsDClient client) {
+
+        this.prefix = client.prefix;
+        this.handler = client.handler;
+        this.constantTagsRendered = client.constantTagsRendered;
+        this.clientChannel = client.clientChannel;
+        this.statsDProcessor = client.statsDProcessor;
+        this.statsDSender = client.statsDSender;
```

Comment:
If I understand correctly the code, as `this.statsDSender = client.statsDSender;` `this.statsDSender` is submitted to `client.executor` and not `this.executor`.
https://github.com/DataDog/java-dogstatsd-client/blob/289edfd19fa457ee4abf497b1a74889733e6d18b/src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java#L233-L234

So https://github.com/DataDog/java-dogstatsd-client/blob/289edfd19fa457ee4abf497b1a74889733e6d18b/src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java#L835 may not shutdown `statsDSender`.

Reply:
Indeed this shallow copy is not good enough, good catch. This must be fixed.